File: src/main/java/ru/bkmz/util/gui/window/Notification.java
package ru.bkmz.util.gui.window;

import javafx.application.Platform;
import javafx.scene.control.Alert;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import static ru.bkmz.util.Decoder.UTFtoWin1251;

public class Notification {
    public static final Logger logger = LogManager.getLogger();

    public Notification(String name, String info) {
        Platform.runLater(new Runnable() {
            @Override
            public void run() {
                logger.info(UTFtoWin1251(name + ":" + info));
                Alert alert = new Alert(Alert.AlertType.INFORMATION);
                alert.setTitle(name);
                alert.setHeaderText(null);
                alert.setContentText(info);
                alert.showAndWait();
            }
        });
    }

    public Notification(String name, String info, Alert.AlertType alert) {
        Platform.runLater(new Runnable() {
            @Override
            public void run() {
                logger.info(UTFtoWin1251(name + ":" + info + ":" + alert));
                Alert nAlert = new Alert(alert);
                nAlert.setTitle(name);
                nAlert.setHeaderText(null);
                nAlert.setContentText(info);
                nAlert.showAndWait();
            }
        });

    }
}

File: src/main/kotlin/ru/bkmz/util/gui/window/Notification.kt
package ru.bkmz.util.gui.window

import javafx.application.Platform
import javafx.scene.control.Alert
import javafx.scene.control.Alert.AlertType
import org.apache.logging.log4j.LogManager
import ru.bkmz.util.Decoder.UTFtoWin1251


class Notification {
    constructor(name: String, info: String) {
        Platform.runLater {
            logger.info(UTFtoWin1251("$name:$info"))
            val alert = Alert(AlertType.INFORMATION)
            alert.title = name
            alert.headerText = null
            alert.contentText = info
            alert.showAndWait()
        }
    }

    constructor(name: String, info: String, alert: AlertType) {
        Platform.runLater {
            logger.info(UTFtoWin1251("$name:$info:$alert"))
            val nAlert = Alert(alert)
            nAlert.title = name
            nAlert.headerText = null
            nAlert.contentText = info
            nAlert.showAndWait()
            nAlert.buttonTypes.removeAll()
        }
    }

    companion object {
        val logger = LogManager.getLogger()
    }
}
